atomic_reactor/plugins/pre_add_labels_in_df.py

```diff
@@ -319,29 +317,21 @@ def escape(s):
 
         labels = []
         for key, value in self.labels.items():
-            try:
-                base_image_value = base_image_labels[key]
-            except KeyError:
-                self.log.info("label %r not present in base image", key)
-            else:
-                # Some labels (self.rewrite_whitelist) should always be set in the dockerfile
-                # even if it was already set in base image and the values match
-                if base_image_value == value and key not in self.rewrite_whitelist:
-                    self.log.info("label %r is already set to %r", key, value)
-                    continue
-                else:
-                    self.log.info("base image has label %r set to %r", key, base_image_value)
-                    if key in self.dont_overwrite:
-                        self.log.info("denying overwrite of label %r", key)
-                        continue
 
-            if (key in self.dont_overwrite_if_in_dockerfile) and (key in dockerfile.labels):
-                self.log.info("denying overwrite of label %r, using from Dockerfile", key)
-                continue
+            if key not in dockerfile.labels or dockerfile.labels[key] != value:
 
-            label = '"%s"="%s"' % (escape(key), escape(value))
-            self.log.info("setting label %r", label)
-            labels.append(label)
+                if key in self.dont_overwrite_if_in_dockerfile and key in dockerfile.labels:
```

Comment:
first we are checking IF we DONT have label in dockerfile OR value is different
in both cases we will be adding / overwriting 

2nd check is IF label is in dont_overwrite_if_in_dockerfile AND the label actually exists in dockerfile